lib/elixir/lib/string.ex

```diff
@@ -1812,4 +1812,91 @@ defmodule String do
       {comm + 1, trans, current}
     end
   end
+
+  @spec difference(t, t) :: [{:eq | :ins | :del, t}] | nil
+  def difference(str1, str2) do
```

Comment:
I've chosen `difference` as it's very common for Elixir to have fully spelled names (I wish `head` and `tail` would exist), for example, it goes nicely along `equivalent?` function. 

> `diff` may identify what this does in an easier way (as it's a known concept)

As you may see linked paper use diff in text only once for UNIX command.
Windows users, in turn, could wonder why we not called this function some other name.